pkg/utils/subscription.go

```diff
@@ -978,3 +981,29 @@ func IsSubscriptionBeDeleted(clt client.Client, subKey types.NamespacedName) boo
 
 	return !subIns.GetDeletionTimestamp().IsZero()
 }
+
+// IsHub determines the hub cluster by listing multiclusterhubs resource items
+func IsHub(config *rest.Config) bool {
+	var dl dynamic.ResourceInterface
+
+	multiclusterHubGVR := schema.GroupVersionResource{
+		Group:    "operator.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "multiclusterhubs",
+	}
+
+	dynamicClient := dynamic.NewForConfigOrDie(config)
+
+	dl = dynamicClient.Resource(multiclusterHubGVR)
+
+	objlist, err := dl.List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Infof("Listing multiclusterHub resource failed, err: %v", err)
+		return false
+	}
+
+	objCount := len(objlist.Items)
```

Comment:
Is it possible to run into a nil pointer here where objlist is nil? Could you please double check?